src/effect.h

```diff
@@ -296,6 +300,11 @@ class effect
             return eff_type->id;
         }
 
+        /** Returns the creature that caused the effect. */
+        const effect_source &get_source() const;
+        /** Sets the creature that caused the effect. */
+        void set_source( const effect_source &source );
```

Comment:
It doesn't look like you're ever calling this function.  Better not to have it if its unused; I feel like it should never be necessary to change the source of an existing effect.